pmd-core/src/main/java/net/sourceforge/pmd/RuleContext.java

```diff
@@ -97,21 +97,29 @@ public void setSourceCodeFile(File sourceCodeFile) {
 
     /**
      * Get the file name associated with the current source file.
+     * If there is no source file, then an empty string is returned.
      *
      * @return The file name.
      */
     public String getSourceCodeFilename() {
-        return sourceCodeFilename;
+        if (sourceCodeFile != null) {
+            return sourceCodeFile.getName();
+        }
+        return "";
     }
 
     /**
      * Set the file name associated with the current source file.
      *
      * @param filename
      *            The file name.
+     * @deprecated This method will be removed. The file should only be
+     * set with {@link #setSourceCodeFile(File)}. Setting the filename here
+     * has no effect.
      */
+    @Deprecated
     public void setSourceCodeFilename(String filename) {
-        this.sourceCodeFilename = filename;
+        // ignored, does nothing.
```

Comment:
we could nag (log an error) about the usage of a deprecated API

Reply:
Yes, makes sense. Internally we don't seem to use it anymore.